src/components/Quests/Quest.js

```diff
@@ -0,0 +1,199 @@
+import { useState, createContext } from 'react'
+import {
+	Flex,
+	Box,
+	Text,
+	IconButton,
+	Image,
+	Stack,
+	Heading,
+} from '@chakra-ui/react'
+import { Link } from 'react-router-dom'
+import {
+	IconCurrent,
+	IconDone,
+	IconLocked,
+	IconCurrentSmall,
+	IconDoneSmall,
+	IconLockedSmall,
+} from './Icons'
+import PathImage from './icons/path.png'
+import { questStatus } from './GlobalQuest'
+
+export const CurrentQuest = createContext()
+
+export default function QuestList() {
+	const [quests, setQuests] = useState([
+		{
+			id: 1,
+			name: 'quest1',
+			isComplete: false,
+			isUnlocked: true,
+		},
+		{ id: 2, name: 'quest2', isComplete: false, isUnlocked: false },
+	])
+
+	function completeQuest(questId) {
+		const updatedQuests = quests.map((quest) => {
+			if (quest.id === questId) {
+				return {
+					...quest,
+					isComplete: true,
+				}
+			} else if (quest.id === questId + 1) {
+				questStatus[`quest${questId + 1}`] = true
+				return { ...quest, isUnlocked: true }
+			} else {
+				return quest
+			}
+		})
+		setQuests(updatedQuests)
+
+		localStorage.setItem(`quest-${questId}`, true)
+
+	}
```

Comment:
As we discussed, this logic should be part of the `questsContext`. This file should access it by:

```
const { quests, completeQuest } = useQuests()
```